docs/style-spec/expressions-examples.md

```diff
@@ -0,0 +1,298 @@
+# Practical examples of expressions
+
+
+## Language fallbacks
+
+### Single language labels
+
+Assigning a text field for an English-language style: 
+
+*NOTES:* 
+
+- Currently, the name fields in our VTs include fallbacks to other languages, but we are planning to remove the VT fallbacks so fields only have values if we have data for that name in that language.
+- In this example, `name_latin` is a field that includes the local language name for a feature *only if* the local language is a Latin language.
+
+```
+{
+	"text-field": {
+		"expression": [
+			["case",
+				["has", ["properties"], "name_latin"], ["string", ["get", ["properties"], "name_latin"]],
+				["has", ["properties"], "name_en"], ["string", ["get", ["properties"], "name_en"]],
+				["string", ["get", ["properties"], "name"]] 
+		]
+	}
+}
+```
+
+### Multi language labels
+
+Assigning a text fields for an English-language plus local language style:
+
+Example output values:
+
+- `Happy Garden` (local language is English)
+- `Glücklich Garten` (local language is a non-English Latin language, e.g. German)
+- `Happy Garden \N 快乐花园` (local language is non-Latin, e.g. Chinese, English value exists)
+- `快乐花园` (local language is non-Latin, e.g. Chinese, and English value doesn't exist)
+
+*NOTE:* 
+
+we have not yet determined the strategy we'll take for multi-lingual labels in our data and styles, but this example demonstrates the type of logic that would be required.
+
+```
+{
+	"text-field": {
+		"expression": [
+			"concat",
+			["case",
+				["has", ["properties"], "name_latin"], ["string", ["get", ["properties"], "name_latin"]],
+				["has", ["properties"], "name_en"], ["string", ["get", ["properties"], "name_en"]],
+				["string", ["get", ["properties"], "name"]] 
+			],
+			["case",
+				["&&"
+					["!", ["has", ["properties"], "name_latin"]], 
+					["has", ["properties"], "name_en"]
+				],
+				["concat", "\n ", ["string", ["get", ["properties"], "name"]]],
+				""
+
+			]
+		]
+	}
+}
+```
+
+## Highway shields
+
+Icon assignment based on shield design and reference number length.
+
+Example output value: 
+`us-interstate-3`
+
+```
+{
+	"icon-image": {
+		"expression" : [
+				"concat",["string", ["get", ["properties"], "shield"]], "-", ["string", ["get", ["properties"], "ref"]]
+		]
```

Comment:
oops, no – that property should be `reflen`, which is the same thing as ^, pre-calculated in the data, and with the exception that values of `1` become `2` (because we use the same icon for both). I don't know yet if we would consider dropping this property in favor of using `length` (since we have users of our tiles that don't use Mapbox GL), but I'll try rewriting this expression to use `length` anyway, as an example.